### PR TITLE
handle special cases for span frames min duration

### PR DESCRIPTION
**Pull request by stuartnelson3**
cases:

0: never capture stacktraces
-1: always capture stacktraces

Closes https://github.com/elastic/apm-agent-go/issues/465

### CODE REVIEW

File: span.go [hunk 1]
```diff
@@ -356,9 +356,17 @@ func (s *Span) End() {
...
+	case -1:
+		// Always set stacktrace
 		s.setStacktrace(1)
+	case 0:
+		// If s.stackFramesMinDuration == 0, we never set stacktrace.
+	default:
+		if !s.dropped() && len(s.stacktrace) == 0 &&
+			s.Duration >= s.stackFramesMinDuration {
+			s.setStacktrace(1)
+		}
```
Comment (stuartnelson3) [inline span.go:369]: This could have been squished into a single conditional, and I'm happy to do so. This just seemed to be more easy to read.

Comment (axw) [reply span.go:369]: Looks good to me. Maybe we can revisit when we get to https://github.com/elastic/apm-agent-go/issues/1142

File: span_test.go [hunk 2]
```diff
@@ -196,6 +196,39 @@ func TestStartExitSpan(t *testing.T) {
...
+	tx = tracer.StartTransaction("name", "type")
+	span = tx.StartSpan("span2", "span2", nil)
+	span.End()
+	tx.End()
+
+	tracer.Flush(nil)
+
+	spans = tracer.Payloads().Spans
+	require.Len(t, spans, 1)
+	assert.Len(t, spans[0].Stacktrace, 4)
```
Comment (axw) [inline span_test.go:229]: Maybe just check it's non-empty? The Go runtime or testing package might add more frames in the future.

**Review by axw — APPROVED** [review]

**Comment by axw** [comment]
Verified using `go.elastic.co/apm/v2 v2.0.0-20220209134054-640a916049c9` (current main):

```go
package main

import (
        "time"

        "go.elastic.co/apm/v2"
)

func main() {
        tracer := apm.DefaultTracer()
        defer tracer.Flush(nil)

        tx := tracer.StartTransaction("name", "type")
        defer tx.End()

        span := tx.StartSpan("name", "type", nil)
        time.Sleep(5 * time.Millisecond)
        span.End()
}
```

- By default, the span has a stack trace
- When setting ELASTIC_APM_SPAN_FRAMES_MIN_DURATION=-1ms, the span still has a stack trace
- When setting ELASTIC_APM_SPAN_FRAMES_MIN_DURATION=0ms, the span does not have a stack trace

**Pull request merged**